experiments/multilingual-fineweb2-hq/download_and_tokenize_fineweb2_hq.py

```diff
@@ -0,0 +1,107 @@
+"""
+Tokenizes the Fineweb2-HQ dataset splits.
+
+This module defines a function that returns tokenization steps for each dataset split available in the
+Fineweb2 dataset.
+"""
+
+import os.path
+
+from levanter.store.cache import CacheOptions
+
+from experiments.llama import llama3_tokenizer
+from marin.download.huggingface.download import DownloadConfig
+from marin.download.huggingface.download_hf import download_hf
+from marin.execution.executor import ExecutorStep, executor_main, output_path_of, this_output_path, versioned
+from marin.processing.tokenize import TokenizeConfig, tokenize
+from marin.processing.tokenize.data_configs import TokenizerStep
+
+FINEWEB2_DATASETS = {
+    "arb_Arab": ["arb_Arab/*.parquet"],
+    "rus_Cyrl": ["rus_Cyrl/*.parquet"],
+    "cmn_Hani": ["cmn_Hani/*.parquet"],
+    "deu_Latn": ["deu_Latn/*.parquet"],
+    "spa_Latn": ["spa_Latn/*.parquet"],
+    "jpn_Jpan": ["jpn_Jpan/*.parquet"],
+    "fra_Latn": ["fra_Latn/*.parquet"],
+    "ita_Latn": ["ita_Latn/*.parquet"],
+    "por_Latn": ["por_Latn/*.parquet"],
+    "pol_Latn": ["pol_Latn/*.parquet"],
+    "nld_Latn": ["nld_Latn/*.parquet"],
+    "ind_Latn": ["ind_Latn/*.parquet"],
+    "tur_Latn": ["tur_Latn/*.parquet"],
+    "ces_Latn": ["ces_Latn/*.parquet"],
+    "fas_Arab": ["fas_Arab/*.parquet"],
+    "hun_Latn": ["hun_Latn/*.parquet"],
+    "swe_Latn": ["swe_Latn/*.parquet"],
+    "ell_Grek": ["ell_Grek/*.parquet"],
+    "dan_Latn": ["dan_Latn/*.parquet"],
+    "vie_Latn": ["vie_Latn/*.parquet"],
+}
+
+FINEWEB2_HQ_MIXTURE_WEIGHTS = {  # From https://huggingface.co/datasets/epfml/FineWeb2-HQ
+    "arb_Arab": 94 / 1024,
+    "rus_Cyrl": 1.2,  # TiB
+    "cmn_Hani": 784 / 1024,  # in GiB
```

Comment:
to be clear you mean it's 784 GiB yeah? and are these compressed sizes?

Reply:
typically I used compressed bytes \approx tokens (which works well for english but may or may not work for othre languages...)

Reply:
Made a quick script to approximate tokens per word:https://colab.research.google.com/drive/1SAd9lg2xnD69pzRwbUgLfKpcOx1sE9rk#scrollTo=XoJAhfuJ7CVK